plusdeck/dbus/interface.py

```diff
@@ -0,0 +1,170 @@
+import asyncio
+from typing import Optional, Self
+
+from sdbus import (  # pyright: ignore [reportMissingModuleSource]
+    dbus_method_async,
+    dbus_property_async,
+    dbus_signal_async,
+    DbusInterfaceCommonAsync,
+)
+
+from plusdeck.client import Client, create_connection, Receiver, State
+from plusdeck.config import Config
+
+DBUS_NAME = "org.jfhbrook.plusdeck"
+
+
+async def load_client(config_file: str) -> Client:
+    config: Config = Config.from_file(config_file)
+
+    client = await create_connection(config.port)
+
+    return client
+
+
+class DbusInterface(  # type: ignore
+    DbusInterfaceCommonAsync, interface_name=DBUS_NAME  # type: ignore
+):
+
+    def __init__(self: Self, config_file: str, client: Client) -> None:
+        super().__init__()
+        self._config_file = config_file
+        self._client: Client = client
+        self._client_lock: asyncio.Lock = asyncio.Lock()
+        self._rcv: Optional[Receiver] = None
+        self.subscribe()
+
+    @dbus_property_async("s")
+    def config_file(self: Self) -> str:
+        return self._config_file
+
+    def subscribe(self: Self) -> None:
+        self._subscription = asyncio.create_task(self.subscription())
+
+    async def subscription(self: Self) -> None:
+        if not self._rcv:
+            self._rcv: Optional[Receiver] = await self.client.subscribe()
+
+        while True:
+            if not self._rcv:
+                break
+            try:
+                state: State = await self._rcv.get_state()
```

Comment:
Do I want to filter out subscribe/unsubscribe events?

Reply:
I think I should leave them in for now, for consistency with receivers in the main client.